runtime/runtime/src/lib.rs

```diff
@@ -497,6 +503,12 @@ impl Runtime {
         result.gas_burnt = exec_fee;
         // Executing actions one by one
         for (action_index, action) in action_receipt.actions.iter().enumerate() {
+            let action_hash = create_hash_upgradable(
+                apply_state.current_protocol_version,
+                &receipt.receipt_id,
+                &apply_state.last_block_hash,
+                u64::max_value() - action_index as u64,
```

Comment:
why do we use `u64::max_value() - action_index` instead of `action_index`?

Reply:
Otherwise it would match generated receipt IDs

Reply:
I think originally we were using it for random values. But now we only use to generate unique `data_id` which is used to match data dependencies